project_euler/problem_083/sol1.py

```diff
@@ -0,0 +1,123 @@
+"""
+Project Euler Problem 83: https://projecteuler.net/problem=83
+
+-----------------------------------------------------------------------------
+
+PROBLEM STATEMENT
+
+In the 5 by 5 matrix below, the minimal path sum from the top left to
+the bottom right, by moving left, right, up, and down, is indicated
+in bold red and is equal to 2297.
+
+[131]  673 [234] [103] [18]
+[201] [96] [342]  965  [150]
+ 630   803  746  [422] [111]
+ 537   699  497  [121]  956
+ 865   732  524  [37]  [331]
+
+
+Find the minimal path sum from the top left to the bottom right by
+moving left, right, up, and down in matrix.txt (right click and
+"Save Link/Target As..."), a 31K text file containing an 80 by 80 matrix.
+
+-----------------------------------------------------------------------------
+
+SOLUTION
+
+We can interpret the input matrix as being a undirected graph, where
+each cell is a vertex connected to its neighbours.
+
+The solution is based on the Bellman-Ford Algorithm
+
+Reference: https://en.wikipedia.org/wiki/Bellman%E2%80%93Ford_algorithm
+
+TIME 0.22s
+
+"""
+
```

Comment:
```suggestion
The solution is based on the Bellman-Ford Algorithm

Reference: https://en.wikipedia.org/wiki/Bellman%E2%80%93Ford_algorithm
"""

```

TIME is not needed here, Machines are both low end and high end. Performance differs on different devices.

Reply:
Okay, understood. I'll also have to modify other PR's. Thanks

Reply:
Thanks

Reply:
True as we'll time it in our Travis.